diff --git a/rtl/rgmii_phy_if.v b/rtl/rgmii_phy_if.v
index ca82572..273799b 100644
--- a/rtl/rgmii_phy_if.v
+++ b/rtl/rgmii_phy_if.v
@@ -215,7 +215,7 @@ oddr #(
     .WIDTH(1)
 )
 clk_oddr_inst (
-    .clk(USE_CLK90 == "TRUE" ? clk90 : clk),
+    .clk(clk),
     .d1(rgmii_tx_clk_1),
     .d2(rgmii_tx_clk_2),
     .q(phy_rgmii_tx_clk)
@@ -227,7 +227,7 @@ oddr #(
     .WIDTH(5)
 )
 data_oddr_inst (
-    .clk(clk),
+    .clk(USE_CLK90 == "TRUE" ? ~clk90 : clk),
     .d1({rgmii_txd_1, rgmii_tx_ctl_1}),
     .d2({rgmii_txd_2, rgmii_tx_ctl_2}),
     .q({phy_rgmii_txd, phy_rgmii_tx_ctl})
