diff --git a/src/main/scala/gemmini/GemminiConfigs.scala b/src/main/scala/gemmini/GemminiConfigs.scala
index 573581e..ad116a5 100644
--- a/src/main/scala/gemmini/GemminiConfigs.scala
+++ b/src/main/scala/gemmini/GemminiConfigs.scala
@@ -224,11 +224,11 @@ case class GemminiArrayConfig[T <: Data : Arithmetic, U <: Data, V <: Data](
   val ACC_ROWS        = ACC_BANKS * ACC_BANK_ROWS
   val LOG2_ACC_ROWS   = log2Up(ACC_ROWS)
 
-  val MNK_BYTES                   = Int.MaxValue / DIM  // TODO: upper bound?
+  val MNK_BYTES                   = 0x20000000 / DIM  // TODO: upper bound?
   val LOG2_MNK_BYTES              = log2Up(MNK_BYTES)
   val MNK_BYTES_PER_TILE_ROW      = MNK_BYTES * DIM
   val LOG2_MNK_BYTES_PER_TILE_ROW = log2Up(MNK_BYTES_PER_TILE_ROW)
-  val TILE_IDX                    = MNK_BYTES / (DIM / cisc_dim)
+  val TILE_IDX                    = MNK_BYTES / DIM * cisc_dim
   val LOG2_TILE_IDX               = log2Up(TILE_IDX)
 
   //--------------------------------------------------------------------------
