diff --git a/arch/riscv/Kconfig b/arch/riscv/Kconfig
index 1f02f1556974..bbaa37ab8c68 100644
--- a/arch/riscv/Kconfig
+++ b/arch/riscv/Kconfig
@@ -248,7 +248,6 @@ config ARCH_RV64I
 	select HAVE_FTRACE_MCOUNT_RECORD if !XIP_KERNEL
 	select HAVE_FUNCTION_GRAPH_TRACER
 	select HAVE_FUNCTION_TRACER if !XIP_KERNEL
-	select SWIOTLB if MMU
 
 endchoice
 
diff --git a/drivers/mmc/host/Kconfig b/drivers/mmc/host/Kconfig
index d6144978e32d..2efdb065c812 100644
--- a/drivers/mmc/host/Kconfig
+++ b/drivers/mmc/host/Kconfig
@@ -1084,6 +1084,14 @@ config MMC_SDHCI_AM654
 
 	  If unsure, say N.
 
+config FPGA_AXI_SDC
+	tristate "FPGA AXI SD Card Controller"
+	help
+	  This selects the FPGA AXI SD Card controller. If you have
+	  a FPGA development board with SD or MMC devices, say Y or M here.
+
+	  If unsure, say N.
+
 config MMC_OWL
 	tristate "Actions Semi Owl SD/MMC Host Controller support"
 	depends on HAS_DMA
diff --git a/drivers/mmc/host/Makefile b/drivers/mmc/host/Makefile
index 4e4ceb32c4b4..d67ddd8eec40 100644
--- a/drivers/mmc/host/Makefile
+++ b/drivers/mmc/host/Makefile
@@ -70,6 +70,7 @@ obj-$(CONFIG_MMC_SUNXI)		+= sunxi-mmc.o
 obj-$(CONFIG_MMC_USDHI6ROL0)	+= usdhi6rol0.o
 obj-$(CONFIG_MMC_TOSHIBA_PCI)	+= toshsd.o
 obj-$(CONFIG_MMC_BCM2835)	+= bcm2835.o
+obj-$(CONFIG_FPGA_AXI_SDC)	+= fpga-axi-sdc.o
 obj-$(CONFIG_MMC_OWL)		+= owl-mmc.o
 
 obj-$(CONFIG_MMC_REALTEK_PCI)	+= rtsx_pci_sdmmc.o
diff --git a/drivers/net/ethernet/Kconfig b/drivers/net/ethernet/Kconfig
index 955abbc5490e..d2fcfeee0c16 100644
--- a/drivers/net/ethernet/Kconfig
+++ b/drivers/net/ethernet/Kconfig
@@ -159,6 +159,14 @@ config ETHOC
 	help
 	  Say Y here if you want to use the OpenCores 10/100 Mbps Ethernet MAC.
 
+config FPGA_AXI_ETH
+	tristate "FPGA AXI Ethernet controller"
+	help
+	  This selects the FPGA AXI Ethernet controller. If you have
+	  a FPGA development board with Ethernet phy, say Y or M here.
+
+	  If unsure, say N.
+
 source "drivers/net/ethernet/packetengines/Kconfig"
 source "drivers/net/ethernet/pasemi/Kconfig"
 source "drivers/net/ethernet/pensando/Kconfig"
diff --git a/drivers/net/ethernet/Makefile b/drivers/net/ethernet/Makefile
index 9eb01169957f..08a5461e30fc 100644
--- a/drivers/net/ethernet/Makefile
+++ b/drivers/net/ethernet/Makefile
@@ -72,6 +72,7 @@ obj-$(CONFIG_NET_VENDOR_NVIDIA) += nvidia/
 obj-$(CONFIG_LPC_ENET) += nxp/
 obj-$(CONFIG_NET_VENDOR_OKI) += oki-semi/
 obj-$(CONFIG_ETHOC) += ethoc.o
+obj-$(CONFIG_FPGA_AXI_ETH) += fpga-axi-eth.o
 obj-$(CONFIG_NET_VENDOR_PACKET_ENGINES) += packetengines/
 obj-$(CONFIG_NET_VENDOR_PASEMI) += pasemi/
 obj-$(CONFIG_NET_VENDOR_QLOGIC) += qlogic/
diff --git a/drivers/tty/serial/Kconfig b/drivers/tty/serial/Kconfig
index a452748c69b2..1051f977a0e5 100644
--- a/drivers/tty/serial/Kconfig
+++ b/drivers/tty/serial/Kconfig
@@ -538,6 +538,32 @@ config SERIAL_UARTLITE_NR_UARTS
 	  Set this to the number of uartlites in your system, or the number
 	  you think you might implement.
 
+config SERIAL_AXI_UART
+       tristate "FPGA AXI serial port support"
+       depends on HAS_IOMEM
+       select SERIAL_CORE
+       help
+         Say Y here if you want to use the FPGA AXI serial controller.
+
+config SERIAL_AXI_UART_CONSOLE
+       bool "Support for console on FPGA AXI serial port"
+       depends on SERIAL_AXI_UART=y
+       select SERIAL_CORE_CONSOLE
+       select SERIAL_EARLYCON
+       help
+         Say Y here if you wish to use a FPGA AXI as the system
+         console (the system console is the device which receives all kernel
+         messages and warnings and which allows logins in single user mode).
+
+config SERIAL_AXI_UART_PORTS
+       int "Maximum number of FPGA AXI serial ports"
+       depends on SERIAL_AXI_UART
+       range 1 256
+       default 1
+       help
+         Set this to the number of FPGA AXI UARTs in your system, or the number
+         you think you might implement.
+
 config SERIAL_SUNCORE
 	bool
 	depends on SPARC
diff --git a/drivers/tty/serial/Makefile b/drivers/tty/serial/Makefile
index 61cc8de95571..af6044473f93 100644
--- a/drivers/tty/serial/Makefile
+++ b/drivers/tty/serial/Makefile
@@ -54,6 +54,7 @@ obj-$(CONFIG_SERIAL_TXX9) += serial_txx9.o
 obj-$(CONFIG_SERIAL_VR41XX) += vr41xx_siu.o
 obj-$(CONFIG_SERIAL_ATMEL) += atmel_serial.o
 obj-$(CONFIG_SERIAL_UARTLITE) += uartlite.o
+obj-$(CONFIG_SERIAL_AXI_UART) += fpga-axi-uart.o
 obj-$(CONFIG_SERIAL_MSM) += msm_serial.o
 obj-$(CONFIG_SERIAL_QCOM_GENI) += qcom_geni_serial.o
 obj-$(CONFIG_SERIAL_OMAP) += omap-serial.o
