diff --git a/drivers/mmc/host/Kconfig b/drivers/mmc/host/Kconfig
index 5e19a961c34d..3b89ed391688 100644
--- a/drivers/mmc/host/Kconfig
+++ b/drivers/mmc/host/Kconfig
@@ -1097,6 +1097,14 @@ config MMC_SDHCI_AM654
 
 	  If unsure, say N.
 
+config FPGA_AXI_SDC
+	tristate "FPGA AXI SD Card Controller"
+	help
+	  This selects the FPGA AXI SD Card controller. If you have
+	  a FPGA development board with SD or MMC devices, say Y or M here.
+
+	  If unsure, say N.
+
 config MMC_OWL
 	tristate "Actions Semi Owl SD/MMC Host Controller support"
 	depends on HAS_DMA
diff --git a/drivers/mmc/host/Makefile b/drivers/mmc/host/Makefile
index ba0c6d0cd85d..8d732823b9d6 100644
--- a/drivers/mmc/host/Makefile
+++ b/drivers/mmc/host/Makefile
@@ -70,6 +70,7 @@ obj-$(CONFIG_MMC_SUNXI)		+= sunxi-mmc.o
 obj-$(CONFIG_MMC_USDHI6ROL0)	+= usdhi6rol0.o
 obj-$(CONFIG_MMC_TOSHIBA_PCI)	+= toshsd.o
 obj-$(CONFIG_MMC_BCM2835)	+= bcm2835.o
+obj-$(CONFIG_FPGA_AXI_SDC)	+= fpga-axi-sdc.o
 obj-$(CONFIG_MMC_OWL)		+= owl-mmc.o
 
 obj-$(CONFIG_MMC_REALTEK_PCI)	+= rtsx_pci_sdmmc.o
diff --git a/drivers/net/ethernet/Kconfig b/drivers/net/ethernet/Kconfig
index 1917da784191..153ad7be135d 100644
--- a/drivers/net/ethernet/Kconfig
+++ b/drivers/net/ethernet/Kconfig
@@ -161,6 +161,14 @@ config ETHOC
 	help
 	  Say Y here if you want to use the OpenCores 10/100 Mbps Ethernet MAC.
 
+config FPGA_AXI_ETH
+	tristate "FPGA AXI Ethernet controller"
+	help
+	  This selects the FPGA AXI Ethernet controller. If you have
+	  a FPGA development board with Ethernet phy, say Y or M here.
+
+	  If unsure, say N.
+
 source "drivers/net/ethernet/packetengines/Kconfig"
 source "drivers/net/ethernet/pasemi/Kconfig"
 source "drivers/net/ethernet/pensando/Kconfig"
diff --git a/drivers/net/ethernet/Makefile b/drivers/net/ethernet/Makefile
index 0d872d4efcd1..d1412a971926 100644
--- a/drivers/net/ethernet/Makefile
+++ b/drivers/net/ethernet/Makefile
@@ -73,6 +73,7 @@ obj-$(CONFIG_NET_VENDOR_NVIDIA) += nvidia/
 obj-$(CONFIG_LPC_ENET) += nxp/
 obj-$(CONFIG_NET_VENDOR_OKI) += oki-semi/
 obj-$(CONFIG_ETHOC) += ethoc.o
+obj-$(CONFIG_FPGA_AXI_ETH) += fpga-axi-eth.o
 obj-$(CONFIG_NET_VENDOR_PACKET_ENGINES) += packetengines/
 obj-$(CONFIG_NET_VENDOR_PASEMI) += pasemi/
 obj-$(CONFIG_NET_VENDOR_QLOGIC) += qlogic/
diff --git a/drivers/tty/serial/Kconfig b/drivers/tty/serial/Kconfig
index 8cbf73f86059..2fb9da1f9cfd 100644
--- a/drivers/tty/serial/Kconfig
+++ b/drivers/tty/serial/Kconfig
@@ -540,6 +540,32 @@ config SERIAL_UARTLITE_NR_UARTS
 	  Set this to the number of uartlites in your system, or the number
 	  you think you might implement.
 
+config SERIAL_AXI_UART
+       tristate "FPGA AXI serial port support"
+       depends on HAS_IOMEM
+       select SERIAL_CORE
+       help
+         Say Y here if you want to use the FPGA AXI serial controller.
+
+config SERIAL_AXI_UART_CONSOLE
+       bool "Support for console on FPGA AXI serial port"
+       depends on SERIAL_AXI_UART=y
+       select SERIAL_CORE_CONSOLE
+       select SERIAL_EARLYCON
+       help
+         Say Y here if you wish to use a FPGA AXI as the system
+         console (the system console is the device which receives all kernel
+         messages and warnings and which allows logins in single user mode).
+
+config SERIAL_AXI_UART_PORTS
+       int "Maximum number of FPGA AXI serial ports"
+       depends on SERIAL_AXI_UART
+       range 1 256
+       default 1
+       help
+         Set this to the number of FPGA AXI UARTs in your system, or the number
+         you think you might implement.
+
 config SERIAL_SUNCORE
 	bool
 	depends on SPARC
diff --git a/drivers/tty/serial/Makefile b/drivers/tty/serial/Makefile
index 238a9557b487..554bb2f5ea78 100644
--- a/drivers/tty/serial/Makefile
+++ b/drivers/tty/serial/Makefile
@@ -53,6 +53,7 @@ obj-$(CONFIG_SERIAL_JSM) += jsm/
 obj-$(CONFIG_SERIAL_TXX9) += serial_txx9.o
 obj-$(CONFIG_SERIAL_ATMEL) += atmel_serial.o
 obj-$(CONFIG_SERIAL_UARTLITE) += uartlite.o
+obj-$(CONFIG_SERIAL_AXI_UART) += fpga-axi-uart.o
 obj-$(CONFIG_SERIAL_MSM) += msm_serial.o
 obj-$(CONFIG_SERIAL_QCOM_GENI) += qcom_geni_serial.o
 obj-$(CONFIG_SERIAL_OMAP) += omap-serial.o
